src/events/mqtt.js

```diff
@@ -1,8 +1,68 @@
 const shimmer = require('shimmer');
+const uuid4 = require('uuid4');
 const tracer = require('../tracer.js');
 const eventInterface = require('../event.js');
 const moduleUtils = require('./module_utils.js');
 
+/**
+ *  Build pub sub parameters.
+ * @param {Object} options options object (optional).
+ * @param {Function} callback callback function (optional).
+ * @returns {Object} Wrapped object.
+ */
+function getPubSubParams(options, callback) {
+    let internalCallback = callback;
+    let internalOptions = options;
+    if (typeof internalOptions === 'function') {
+        internalCallback = options;
+        internalOptions = {};
+    }
+
+    return { internalCallback, internalOptions };
+}
+
+/**
+ * Set hidden epsagon id in user props.
+ * @param {*} epsagonId - epsagon identifier.
+ * @param {*} message - mqtt message property.
+ * @returns {Object} options with epsagon id.
+ */
+function setEpsagonIdToMessage(epsagonId, message) {
+    if (typeof message === 'object') {
+        return { ...message, epsagonId };
+    } if (typeof message === 'string') {
+        try {
+            const resultMsg =
+            { ...JSON.parse(message), epsagonId };
```

Comment:
I think `JSON.parse(message)` can sometimes return undefined when it fails instead of throwing and error, maybe let's validate that (also it can be a json dumped string, i.e.: `""this string""` where we do not want to add the id) by making sure after parsing that we got an object